types/compat/re.pp

```diff
@@ -0,0 +1,3 @@
+# Emulate the validate_re function
+# validate_re(value, re) translates to Pattern[re], which is not valid puppet and can't be aliased.
```

Comment:
Yup, that is a very unfortunate choice of words. I'll fix that to ".. which is not directly mappable as a type alias, but can be specified as Pattern[re]."